parser_library/src/context/instruction.cpp

```diff
@@ -510,1418 +561,1417 @@ constexpr auto VRV_3 = instruction_format_definition_factory<mach_format::VRV, v
-    { "VSCBI", VRR_c_4, 1581 },
-    { "VSCEF", VRV_3, 1548 },
-    { "VSCEG", VRV_3, 1548 },
-    { "VSD", RI_a_2_u, 0 },
-    { "VSD", RI_a_2_u, 0 },
-    { "VSDP", VRI_f_5, 1656 },
-    { "VSDS", RI_a_2_u, 0 },
-    { "VSDS", RI_a_2_u, 0 },
-    { "VSE", RI_a_2_u, 0 },
-    { "VSEG", VRR_a_3, 1549 },
-    { "VSEL", VRR_e_4, 1549 },
-    { "VSES", RI_a_2_u, 0 },
-    { "VSL", VRR_c_3, 1579 },
-    { "VSLB", VRR_c_3, 1579 },
-    { "VSLD", VRI_d_4, 1607 },
-    { "VSLDB", VRI_d_4, 1579 },
-    { "VSLL", RRE_2, 0 },
-    { "VSP", VRI_f_5, 1658 },
-    { "VSPSD", RRE_2, 0 },
-    { "VSRA", VRR_c_3, 1579 },
-    { "VSRAB", VRR_c_3, 1580 },
-    { "VSRD", VRI_d_4, 1608 },
-    { "VSRL", VRR_c_3, 1580 },
-    { "VSRLB", VRR_c_3, 1580 },
-    { "VSRP", VRI_g_5_s, 1657 },
-    { "VSRRS", RRE_2, 0 },
-    { "VSRSV", RRE_2, 0 },
-    { "VSS", RI_a_2_u, 0 },
-    { "VST", VRX_3_opt, 1550 },
-    { "VSTBR", VRX_3, 1576 },
-    { "VSTD", RI_a_2_u, 0 },
-    { "VSTEB", VRX_3, 1550 },
-    { "VSTEBRF", VRX_3, 1576 },
-    { "VSTEBRG", VRX_3, 1576 },
-    { "VSTEBRH", VRX_3, 1576 },
-    { "VSTEF", VRX_3, 1550 },
-    { "VSTEG", VRX_3, 1550 },
-    { "VSTEH", VRX_3, 1550 },
-    { "VSTER", VRX_3, 1578 },
-    { "VSTH", RI_a_2_u, 0 },
-    { "VSTI", RRE_2, 0 },
-    { "VSTID", RRE_2, 0 },
-    { "VSTK", RI_a_2_u, 0 },
-    { "VSTKD", RI_a_2_u, 0 },
-    { "VSTL", VRS_b_3, 1552 },
-    { "VSTM", VRS_a_4_opt, 1551 },
-    { "VSTMD", RI_a_2_u, 0 },
-    { "VSTRC", VRR_d_6_opt, 1590 },
-    { "VSTRL", VSI_3, 1551 },
-    { "VSTRLR", VRS_d_3, 1551 },
-    { "VSTRS", VRR_d_6_opt, 1622 },
-    { "VSTVM", RRE_2, 0 },
-    { "VSTVP", RRE_2, 0 },
-    { "VSUM", VRR_c_4, 1583 },
-    { "VSUMG", VRR_c_4, 1582 },
-    { "VSUMQ", VRR_c_4, 1583 },
-    { "VSVMM", RRE_2, 0 },
-    { "VTM", VRR_a_2, 1584 },
-    { "VTP", VRR_g_1, 1660 },
-    { "VTVM", RRE_2, 0 },
-    { "VUPH", VRR_a_3, 1552 },
-    { "VUPKZ", VSI_3, 1660 },
-    { "VUPL", VRR_a_3, 1553 },
-    { "VUPLH", VRR_a_3, 1553 },
-    { "VUPLL", VRR_a_3, 1554 },
-    { "VX", VRR_c_3, 1565 },
-    { "VXELD", RRE_2, 0 },
-    { "VXELE", RRE_2, 0 },
-    { "VXS", RI_a_2_u, 0 },
-    { "VXVC", RRE_2, 0 },
-    { "VXVM", RRE_2, 0 },
-    { "VXVMM", RRE_2, 0 },
-    { "VZPSD", RRE_2, 0 },
-    { "WFC", VRR_a_4, 1599 },
-    { "WFK", VRR_a_4, 1600 },
-    { "X", RX_a_2_ux, 738 },
-    { "XC", SS_a_2_s, 739 },
-    { "XG", RXY_a_2, 738 },
-    { "XGR", RRE_2, 738 },
-    { "XGRK", RRF_a_3, 738 },
-    { "XI", SI_2_u, 739 },
-    { "XIHF", RIL_a_2, 740 },
-    { "XILF", RIL_a_2, 740 },
-    { "XIY", SIY_2_su, 739 },
-    { "XR", RR_2, 738 },
-    { "XRK", RRF_a_3, 738 },
-    { "XSCH", S_0, 1215 },
-    { "XY", RXY_a_2, 738 },
-    { "ZAP", SS_b_2, 928 },
+constexpr std::pair<machine_instruction, supported_system> machine_instructions[] = {
+    { { "A", RX_a_2_ux }, { supported_system::UNI | supported_system::ESA | supported_system::XA | supported_system::_370 | supported_system::DOS | supported_system::SINCE_ZOP } }, 
```

Comment:
There is is <20 of valid combinations, I suggest defining them as individual constexpr values to make this a bit more easier to read.

Reply:
Good idea